R/data.table.R

```diff
@@ -147,6 +147,365 @@ replace_dot_alias = function(e) {
   }
 }
 
+# Transform lapply(.SD, fun) or Map(fun, .SD) into list(fun(col1), fun(col2), ...)
+.massageSD = function(jsub, sdvars, SDenv, funi) {
+  txt = as.list(jsub)[-1L]
+  if (length(names(txt))>1L) .Call(Csetcharvec, names(txt), 2L, "")  # fixes bug #110
+  # support Map instead of lapply #5336
+  fun = if (jsub %iscall% "Map") txt[[1L]] else txt[[2L]]
+  if (fun %iscall% "function") {
+    # Fix for #2381: added SDenv$.SD to 'eval' to take care of cases like: lapply(.SD, function(x) weighted.mean(x, bla)) where "bla" is a column in DT
+    # http://stackoverflow.com/questions/13441868/data-table-and-stratified-means
+    # adding this does not compromise in speed (that is, not any lesser than without SDenv$.SD)
+    # replaced SDenv$.SD to SDenv to deal with Bug #87 reported by Ricardo (Nice catch!)
+    thisfun = paste0("..FUN", funi) # Fix for #985
```

Comment:
If this function raises and error then it can look cryptic. Maybe let's give it a longer more verbose name, or even reuse original name with a prefix?